src/components/Wizard/Wizard.tsx

```diff
@@ -119,61 +112,87 @@ export default function Wizard<T>({
 		resetAndClose();
 	}
 
+	function confirmTermination() {
```

Comment:
To me it seems like this is a perfect candidate for a custom hook.
The component logic is now getting a tad complex as well. A custom hook would prevent that and will keep testing easier.
`useConfirmTermination()` ftw?

Reply:
Moved to hook and added tests